### PR TITLE
test: Bump MAX_NODES to 12

### PR DESCRIPTION
**Pull request by maflcko**
When testing a combination of settings that affect the datadir (e.g. prune, blockfilter, ...) we may need a lot of datadirs. 
Bump the maximum number of nodes proactively from 8 to 12, so that caches get populated with 12 node dirs, as opposed to 8.

Also, add an assert that the list of deterministic keys is exactly the number of max nodes (and not more than that.

Also, create the cache faster.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Added a commit to create the cache in less than a second.

Can be tested with `time ./test/functional/create_cache.py --cachedir="/tmp/test/cache_$(date)" --tmpdir=/tmp/test/temp`

**Comment by laanwj** [comment]
utACK fa473303972b7dad600d949dc9b303d8136cb7e7
I think tests should create as few nodes as they can get away with, but if a test with 12 nodes is useful to get more coverage, then let's bump this.

**Comment by laanwj** [comment]
> When testing a combination of settings that affect the datadir (e.g. prune, blockfilter, ...) we may need a lot of datadirs.

Oh so these run in serial instead of in parallel? That makes more sense then.

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -124,8 +123,15 @@ def get_deterministic_priv_key(self):
             AddressKeyPair('myzuPxRwsf3vvGzEuzPfK9Nf2RfwauwYe6', 'cQMpDLJwA8DBe9NcQbdoSb1BhmFxVjWD5gRyrLZCtpuF9Zi3a9RK'),
             AddressKeyPair('mumwTaMtbxEPUswmLBBN3vM9oGRtGBrys8', 'cSXmRKXVcoouhNNVpcNKFfxsTsToY5pvB9DVsFksF1ENunTzRKsy'),
             AddressKeyPair('mpV7aGShMkJCZgbW7F6iZgrvuPHjZjH9qg', 'cSoXt6tm3pqy43UMabY6eUTmR3eSUYFtB2iNQDGgb3VUnRsQys2k'),
-        ]
-        return PRIV_KEYS[self.index]
+            AddressKeyPair('mq4fBNdckGtvY2mijd9am7DRsbRB4KjUkf', 'cN55daf1HotwBAgAKWVgDcoppmUNDtQSfb7XLutTLeAgVc3u8hik'),
+            AddressKeyPair('mpFAHDjX7KregM3rVotdXzQmkbwtbQEnZ6', 'cT7qK7g1wkYEMvKowd2ZrX1E5f6JQ7TM246UfqbCiyF7kZhorpX3'),
```
Comment (decryp2kanon) [inline test/functional/test_framework/test_node.py:127]: how to create these address and privkey? it seems have no prefix. just curious!

**Pull request merged**